ocamldoc/Makefile

```diff
@@ -44,24 +44,25 @@ OCAMLPP=-pp 'sh ./remove_DEBUG'
 MKDIR=mkdir -p
 CP=cp
 OCAMLDOC=ocamldoc
+OCAMLDOC_OPT=$(OCAMLDOC).opt
 
-# TODO: clarify whether the following really needs to be that complicated
 ifeq "$(UNIX_OR_WIN32)" "unix"
-  ifeq "$(TARGET)" "$(HOST)"
-    ifeq "$(SUPPORTS_SHARED_LIBRARIES)" "true"
-      OCAMLDOC_RUN=$(OCAMLRUN) -I $(ROOTDIR)/otherlibs/$(UNIXLIB) -I $(ROOTDIR)/otherlibs/str ./$(OCAMLDOC)
-    else
-      OCAMLDOC_RUN=./$(OCAMLDOC)
-    endif
-  else
-    OCAMLDOC_RUN=$(OCAMLRUN) ./$(OCAMLDOC)
-  endif
+  OCAMLDOC_RUN_BYTE=$(OCAMLRUN) -I $(ROOTDIR)/otherlibs/$(UNIXLIB) -I $(ROOTDIR)/otherlibs/str ./$(OCAMLDOC)
```

Comment:
This breaks `make world` on a `--disable-shared` build. The issue is that when shared support isn't available, `ocamldoc` is built with `-custom` so mustn't be invoked via `ocamlrun`

Reply:
Note: I can't find where in the Makefile logic `-custom` is used in absence of shared libraries (or at all) to compile ocamldoc. (There is logic to that effect in `Makefile.tools` but as far as I can tell it's not used to build ocamldoc.)

Reply:
It's done using the "lib_custom" flag in and `str.cmo` & `unix.cmo` - on a system without shared linking, using the `.cmo` forces `-custom` mode (see https://github.com/ocaml/ocaml/blob/trunk/file_formats/cmo_format.mli#L55 and https://github.com/ocaml/ocaml/blob/trunk/bytecomp/bytelink.ml#L55)

Reply:
Wow. Thanks. I'll definitely add a comment in the ocamldoc Makefile to point this out.